src/Features/CSharp/Portable/RemoveUnusedVariable/CSharpRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -69,5 +69,18 @@ protected override void RemoveOrReplaceNode(SyntaxEditor editor, SyntaxNode node
 
         protected override SeparatedSyntaxList<SyntaxNode> GetVariables(LocalDeclarationStatementSyntax localDeclarationStatement)
             => localDeclarationStatement.Declaration.Variables;
+
+        protected override bool ShouldOfferFixForLocalDeclaration(ISyntaxFactsService syntaxFacts, LocalDeclarationStatementSyntax localDeclaration)
+        {
+            // Local declarations must be parented by an executable block, or global statement, otherwise
+            // removing them would be invalid (and more than likely crash the fixer)
+            if (localDeclaration.Parent is not GlobalStatementSyntax &&
+                !syntaxFacts.IsExecutableBlock(localDeclaration.Parent))
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
nit, consider:

```suggestion
            return localDeclaration.Parent is GlobalStatementSyntax ||
                syntaxFacts.IsExecutableBlock(localDeclaration.Parent))
```

This is simpler or me as there's a lot less negations.  in the original there are two negations in the condition itself.  and then the truthy condition is negated for a falsy result.  Personally, i find that more confusing than just returning this simpler truthy expression.  Up to you though.